File: 1275-find-winner-on-a-tic-tac-toe-game/1275-find-winner-on-a-tic-tac-toe-game.js
/**
 * @param {number[][]} moves
 * @return {string}
 */
var tictactoe = function(moves) {
    let grid = Array(3).fill('').map(()=>Array(3).fill(''));
    let aTurn = true;
    for (let move of moves) {
        const [x, y] = move;
        if (aTurn) {
            grid[x][y] = 'X';
        } else {
            grid[x][y] = 'O';
        }
        aTurn = !aTurn;
    }   
    console.log(grid)
    return winnerCheck(grid);
};

var winnerCheck = function(grid) {    
    //Check for winner in all rows
    for (let row of grid) {
        if (row.every(val => val === 'X')) {
            return "A"
        }
        if (row.every(val => val === 'O')) {
            return "B"
        }
    }
    //Check for winner in all columns
    for (let i=0; i<grid.length; i++) {
        let xCount = 0, oCount = 0;
        for (let j=0; j<grid.length; j++) {
            if (grid[j][i] === 'X') {
                xCount++;
            } else if (grid[j][i] === 'O') {
                oCount++;
            }
        }
        if (xCount === 3) return 'A';
        if (oCount === 3) return 'B';
    }
    //Check for winner in diagonal1
    let xCounter = 0, oCounter = 0;
    for (let k=0; k<grid.length; k++) {
        if (grid[k][k] === 'X') {
            xCounter++;
        } else if (grid[k][k] === 'O') {
            oCounter++;
        }
        if (xCounter === 3) return 'A';
        if (oCounter === 3) return 'B';
    }
    
    //Check for winner in diagonal2
    xCounter = 0, oCounter = 0;
    for (let k=0; k<grid.length; k++) {
        if (grid[k][grid.length-1-k] === 'X') {
            xCounter++;
        } else if (grid[k][grid.length-1-k] === 'O') {
            oCounter++;
        }
        if (xCounter === 3) return 'A';
        if (oCounter === 3) return 'B';
    }    
    
    //Check for empty spaces for more movements
    for (let m=0; m<grid.length; m++) {
        for (let n=0; n<grid.length; n++) {
            if (grid[m][n] === '') return 'Pending';
        }
    }
    
    return 'Draw';
}

File: 1275-find-winner-on-a-tic-tac-toe-game/1275-find-winner-on-a-tic-tac-toe-game.py
class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        rows = [0,0,0]
        cols = [0,0,0]
        count = main_diag = anti_diag = 0
        
        for move in moves:
            r, c = move
            # 1 represents 'X' and -1 represents 'O'
            value = 1 if count % 2 == 0 else -1
            rows[r] += value
            cols[c] += value
            
            if r == c:
                main_diag += value
            if r + c == 2:
                anti_diag += value
            
            # Check for winner
            if 3 in [rows[r], cols[c], main_diag, anti_diag]:
                return 'A'
            elif -3 in [rows[r], cols[c], main_diag, anti_diag]:
                return 'B'
            
            count += 1
            print(rows)
            print(cols)
        if count < 9:
            return 'Pending'
        
        return 'Draw'
            